src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -390,42 +387,17 @@ public SearchRequest templateSource(String template) {
     }
 
     /**
-     * The name of the stored template
+     * The stored template
      */
-    public void templateName(String templateName) {
-        this.templateName = templateName;
-    }
-
-    public void templateType(ScriptService.ScriptType templateType) {
-        this.templateType = templateType;
-    }
-
-    /**
-     * Template parameters used for rendering
-     */
-    public void templateParams(Map<String, Object> params) {
-        this.templateParams = params;
+    public void template(Template template) {
+        this.template = template;
     }
 
     /**
-     * The name of the stored template
+     * The stored template
      */
-    public String templateName() {
-        return templateName;
-    }
-
-    /**
-     * The name of the stored template
-     */
-    public ScriptService.ScriptType templateType() {
-        return templateType;
-    }
-
-    /**
-     * Template parameters used for rendering
-     */
-    public Map<String, Object> templateParams() {
-        return templateParams;
```

Comment:
no both Request and RequestBuilders are java api. one can choose which one to use. you can always do client.search(SearchRequest) without going through the builder.